core/client/fs/src/main/java/alluxio/client/file/ufs/UfsFileInStream.java

```diff
@@ -162,6 +170,12 @@ private void updateStreamIfNeeded() {
     if (mUfsInStream.isPresent()) {
       return;
     }
-    mUfsInStream = Optional.of(mFileOpener.apply(mPosition));
+    InputStream ufsInStream = mFileOpener.apply(mPosition);
```

Comment:
I'm still a bit confused why we need to pass fileOpener instead of just have a function inside?

Reply:
File Opener is a function `Function<Long, InputStream>` to open the file.
This is for convenience, or we can pass in `UnderFileSystem` and file path and add a `openFile(offset)` function inside the UfsFileInStream.
Both approaches work, i do not see which one is much better than another
```
      return new UfsFileInStream(offset -> {
        try {
          return mUfs.get().open(status.getPath(), OpenOptions.defaults().setOffset(offset));
        } catch (IOException e) {
          throw AlluxioRuntimeException.from(e);
        }
      }, status.getLength());
    });
```